EventListener/ControllerListener.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Novaway\Bundle\FeatureFlagBundle\EventListener;
+
+use Doctrine\Common\Annotations\Reader;
+use Doctrine\Common\Util\ClassUtils;
+use Novaway\Bundle\FeatureFlagBundle\Annotation\Feature;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class ControllerListener implements EventSubscriberInterface
+{
+    /** @var Reader */
+    private $reader;
+
+    /**
+     * Constructor
+     *
+     * @param Reader $reader
+     */
+    public function __construct(Reader $reader)
+    {
+        $this->reader = $reader;
+    }
+
+    /**
+     * Update the request object to apply annotation configuration
+     *
+     * @param FilterControllerEvent $event
+     */
+    public function onKernelController(FilterControllerEvent $event)
+    {
+        $controller = $event->getController();
+        if (!is_array($controller) && method_exists($controller, '__invoke')) {
+            $controller = array($controller, '__invoke');
+        }
+
+        if (!is_array($controller)) {
+            return;
+        }
+
+        $className = class_exists('Doctrine\Common\Util\ClassUtils') ? ClassUtils::getClass($controller[0]) : get_class($controller[0]);
+        $object    = new \ReflectionClass($className);
+        $method    = $object->getMethod($controller[1]);
+
+        $features = [];
+        foreach ($this->reader->getMethodAnnotations($method) as $annotation) {
+            if ($annotation instanceof Feature) {
+                $key = $annotation->getFeature();
+                if (isset($features[$key])) {
+                    throw new \UnexpectedValueException(sprintf('Feature "%s" is defined more than once for %s::%s', $key, $className, $controller[1]));
+                }
+
+                $features[$key] = [
+                    'feature' => $annotation->getFeature(),
+                    'enabled' => $annotation->getEnabled(),
+                ];
```

Comment:
On [line 36](https://github.com/novaway/NovawayFeatureFlagBundle/pull/4/files#diff-50211849393457d7baea165ea5fddce1R36) the PHP 5.3 syntax is used for arrays. Here we have the 5.4 syntax. 

Either 5.3 compatibility is required and this instruction fails, or there is a code standard issue.

I suggest we update the `composer.json` file with the min PHP supported version to make this clear. My suggestion is to drop 5.3 compatibility.